### PR TITLE
[sharktank] Remove 'torch' from deps and warn instead

### PR DESCRIPTION
**Pull request by marbre**
Instead of enforcing the installation for 'torch' as a dependency, error if 'torch' cannot be imported and point the user to how to install.

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -34,14 +34,18 @@ Setup your Python environment with the following commands:
...
-# Optional: faster installation of torch with just CPU support.
+```bash
+# Fast installation of torch with just CPU support.
 # See other options at https://pytorch.org/get-started/locally/
 pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu
 ```
 
-## Install SHARK and its dependencies
-
 ```bash
```
Comment (ScottTodd) [inline docs/user_guide.md:49]:
```suggestion
Next install shark-ai:

```bash
```

File: docs/user_guide.md [hunk 2]
```diff
@@ -34,14 +34,18 @@ Setup your Python environment with the following commands:
 # Set up a virtual environment to isolate packages from other envs.
 python3.11 -m venv 3.11.venv
 source 3.11.venv/bin/activate
+```
+
+## Install SHARK and its dependencies
+
+Before installing `shark-ai` install a torch version that fulfills your needs.
```
Comment (ScottTodd) [inline docs/user_guide.md:41]:
```suggestion
First install a torch version that fulfills your needs:
```

File: docs/user_guide.md [hunk 3]
```diff
@@ -34,14 +34,18 @@ Setup your Python environment with the following commands:
...
 
-# Optional: faster installation of torch with just CPU support.
+```bash
+# Fast installation of torch with just CPU support.
 # See other options at https://pytorch.org/get-started/locally/
```
Comment (ScottTodd) [inline docs/user_guide.md:45]: I guess we could pull this URL out of the code block so it is clickable 🤔

**Review by ScottTodd — APPROVED** [review]
Structurally LGTM. A few phrasing nits

File: docs/user_guide.md (hunk 3 above)
Comment (marbre) [reply docs/user_guide.md]: Sounds good.

**Pull request merged**